wa/workloads/speedometer/__init__.py

```diff
@@ -12,57 +12,298 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from http.server import SimpleHTTPRequestHandler, HTTPServer
+import logging
 import os
 import re
+import subprocess
+import tarfile
+import threading
+import time
 
-from wa import ApkUiautoWorkload, Parameter
-from wa.framework.exception import ValidationError, WorkloadError
-from wa.utils.types import list_of_strs
-from wa.utils.misc import unique
+from wa import Parameter, Workload
+from wa.framework.exception import WorkloadError
 
+from devlib.utils.android import adb_command
 
-class Speedometer(ApkUiautoWorkload):
 
-    name = 'speedometer'
-    package_names = ['com.android.chrome']
-    regex = re.compile(r'Speedometer Score ([\d.]+)')
-    versions = ['1.0', '2.0']
-    description = '''
-    A workload to execute the speedometer web based benchmark
+class ArchiveServer:
+    def __init__(self):
+        self._port = None
+
+    def start(self, document_root, target):
+        class _DifferentDirectoryHTTPRequestHandler(SimpleHTTPRequestHandler):
```

Comment:
Is there an issue with defining these classes at the top level? Function-local definitions should be avoided in Python -- they case problems.